python3Packages.{gradio: 5.49.1 -> 6.1.0,gradio-client: 1.12.1 -> 2.0.1,gradio-pdf: 0.0.22 -> 0.0.23}

mdaniels5757 opened the pull request:
Closes: #445159

* python3Packages.gradio: add updateScript
* python3Packages.gradio-client: modify updateScript
  This change is needed because nix-update only checks so many versions, so sometimes gradio's releases align such that no version for gradio_client is present in the first N releases.
* python3Packages.gradio-client: 1.12.1 -> 2.0.1
  Despite the pinned `rev`, this is exactly version 2.0.1, not unstable.
 See https://github.com/gradio-app/gradio/pull/12478.
  Upstream issue: https://github.com/gradio-app/gradio/issues/12541.
  Changelog: https://github.com/gradio-app/gradio/blob/7a8894d7249ee20c2f7a896237e290e99661fd43/client/python/CHANGELOG.md
* python3Packages.gradio-pdf: 0.0.22 -> 0.0.23
  Diff: https://github.com/freddyaboulton/gradio-pdf/compare/8833e9cd419d2a5eeff98e3ae8cbe690913bcfce...57deb0bd2823a9a75c5da498cc39a85f8fd4ff02
* python3Packages.gradio: 5.49.1 -> 6.1.0
  Changelog: https://github.com/gradio-app/gradio/blob/gradio%406.1.0/CHANGELOG.md

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [X] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

mdaniels5757 commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 446537`
Commit: [`23e1379750d34168016a09e0bd221b3707db768b`](https://github.com/NixOS/nixpkgs/commit/23e1379750d34168016a09e0bd221b3707db768b) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/23e1379750d34168016a09e0bd221b3707db768b..pull/446537/head))
Merge: [`21cf1dcacd44e9baf706e4b0e07a18746f36cc1a`](https://github.com/NixOS/nixpkgs/commit/21cf1dcacd44e9baf706e4b0e07a18746f36cc1a)

Logs: https://github.com/mdaniels5757/nixpkgs-review-gha/actions/runs/18961313654


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 22 packages built:</summary>
  <ul>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio.dist</li>
    <li>python312Packages.smolagents</li>
    <li>python312Packages.smolagents.dist</li>
    <li>python312Packages.txtai</li>
    <li>python312Packages.txtai.dist</li>
    <li>python313Packages.gradio</li>
    <li>python313Packages.gradio-client</li>
    <li>python313Packages.gradio-client.dist</li>
    <li>python313Packages.gradio-pdf</li>
    <li>python313Packages.gradio-pdf.dist</li>
    <li>python313Packages.gradio.dist</li>
    <li>python313Packages.smolagents</li>
    <li>python313Packages.smolagents.dist</li>
    <li>python313Packages.txtai</li>
    <li>python313Packages.txtai.dist</li>
    <li>rembg</li>
    <li>rembg.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 6 packages failed to build:</summary>
  <ul>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio.dist</li>
    <li>python312Packages.smolagents</li>
    <li>python312Packages.smolagents.dist</li>
    <li>python312Packages.txtai</li>
    <li>python312Packages.txtai.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 14 packages built:</summary>
  <ul>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python313Packages.gradio</li>
    <li>python313Packages.gradio-client</li>
    <li>python313Packages.gradio-client.dist</li>
    <li>python313Packages.gradio-pdf</li>
    <li>python313Packages.gradio-pdf.dist</li>
    <li>python313Packages.gradio.dist</li>
    <li>python313Packages.smolagents</li>
    <li>python313Packages.smolagents.dist</li>
    <li>python313Packages.txtai</li>
    <li>python313Packages.txtai.dist</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 16 packages failed to build:</summary>
  <ul>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio.dist</li>
    <li>python312Packages.smolagents</li>
    <li>python312Packages.smolagents.dist</li>
    <li>python313Packages.gradio</li>
    <li>python313Packages.gradio-client</li>
    <li>python313Packages.gradio-client.dist</li>
    <li>python313Packages.gradio-pdf</li>
    <li>python313Packages.gradio-pdf.dist</li>
    <li>python313Packages.gradio.dist</li>
    <li>python313Packages.smolagents</li>
    <li>python313Packages.smolagents.dist</li>
  </ul>
</details>

---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 16 packages failed to build:</summary>
  <ul>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio.dist</li>
    <li>python312Packages.smolagents</li>
    <li>python312Packages.smolagents.dist</li>
    <li>python313Packages.gradio</li>
    <li>python313Packages.gradio-client</li>
    <li>python313Packages.gradio-client.dist</li>
    <li>python313Packages.gradio-pdf</li>
    <li>python313Packages.gradio-pdf.dist</li>
    <li>python313Packages.gradio.dist</li>
    <li>python313Packages.smolagents</li>
    <li>python313Packages.smolagents.dist</li>
  </ul>
</details>

mdaniels5757 commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 446537`
Commit: [`23e1379750d34168016a09e0bd221b3707db768b`](https://github.com/NixOS/nixpkgs/commit/23e1379750d34168016a09e0bd221b3707db768b) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/23e1379750d34168016a09e0bd221b3707db768b..pull/446537/head))
Merge: [`21cf1dcacd44e9baf706e4b0e07a18746f36cc1a`](https://github.com/NixOS/nixpkgs/commit/21cf1dcacd44e9baf706e4b0e07a18746f36cc1a)

Logs: https://github.com/mdaniels5757/nixpkgs-review-gha/actions/runs/18961342395


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 22 packages built:</summary>
  <ul>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio.dist</li>
    <li>python312Packages.smolagents</li>
    <li>python312Packages.smolagents.dist</li>
    <li>python312Packages.txtai</li>
    <li>python312Packages.txtai.dist</li>
    <li>python313Packages.gradio</li>
    <li>python313Packages.gradio-client</li>
    <li>python313Packages.gradio-client.dist</li>
    <li>python313Packages.gradio-pdf</li>
    <li>python313Packages.gradio-pdf.dist</li>
    <li>python313Packages.gradio.dist</li>
    <li>python313Packages.smolagents</li>
    <li>python313Packages.smolagents.dist</li>
    <li>python313Packages.txtai</li>
    <li>python313Packages.txtai.dist</li>
    <li>rembg</li>
    <li>rembg.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 6 packages failed to build:</summary>
  <ul>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio.dist</li>
    <li>python312Packages.smolagents</li>
    <li>python312Packages.smolagents.dist</li>
    <li>python312Packages.txtai</li>
    <li>python312Packages.txtai.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 14 packages built:</summary>
  <ul>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python313Packages.gradio</li>
    <li>python313Packages.gradio-client</li>
    <li>python313Packages.gradio-client.dist</li>
    <li>python313Packages.gradio-pdf</li>
    <li>python313Packages.gradio-pdf.dist</li>
    <li>python313Packages.gradio.dist</li>
    <li>python313Packages.smolagents</li>
    <li>python313Packages.smolagents.dist</li>
    <li>python313Packages.txtai</li>
    <li>python313Packages.txtai.dist</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = false)
<details>
  <summary>:x: 16 packages failed to build:</summary>
  <ul>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio.dist</li>
    <li>python312Packages.smolagents</li>
    <li>python312Packages.smolagents.dist</li>
    <li>python313Packages.gradio</li>
    <li>python313Packages.gradio-client</li>
    <li>python313Packages.gradio-client.dist</li>
    <li>python313Packages.gradio-pdf</li>
    <li>python313Packages.gradio-pdf.dist</li>
    <li>python313Packages.gradio.dist</li>
    <li>python313Packages.smolagents</li>
    <li>python313Packages.smolagents.dist</li>
  </ul>
</details>

---
### `aarch64-darwin` (sandbox = false)
<details>
  <summary>:x: 16 packages failed to build:</summary>
  <ul>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio.dist</li>
    <li>python312Packages.smolagents</li>
    <li>python312Packages.smolagents.dist</li>
    <li>python313Packages.gradio</li>
    <li>python313Packages.gradio-client</li>
    <li>python313Packages.gradio-client.dist</li>
    <li>python313Packages.gradio-pdf</li>
    <li>python313Packages.gradio-pdf.dist</li>
    <li>python313Packages.gradio.dist</li>
    <li>python313Packages.smolagents</li>
    <li>python313Packages.smolagents.dist</li>
  </ul>
</details>

tobim commented:
There is now https://github.com/gradio-app/gradio/releases/tag/%40gradio%2Fclient%401.19.1 and https://github.com/gradio-app/gradio/releases/tag/gradio%405.49.1.

You now also need to add `pillow` to `pythonRelaxDeps`.

mdaniels5757 committed:
python3Packages.gradio-client: modify updateScript

This change is needed because nix-update only checks so
many versions, so sometimes gradio's releases align such that no
version for gradio_client is present in the first N releases.

mdaniels5757 committed:
python3Packages.gradio: add updateScript

mdaniels5757 committed:
python3Packages.gradio-client: 1.12.1 -> 2.0.1

Despite the pinned `rev`, this is exactly version 2.0.1, not unstable.
See gradio-app/gradio#12478.

Upstream issue: gradio-app/gradio#12541.

Changelog: https://github.com/gradio-app/gradio/blob/7a8894d7249ee20c2f7a896237e290e99661fd43/client/python/CHANGELOG.md

mdaniels5757 commented:
Thanks for the update! I've updated `gradio` and added to the pythonRelaxDeps. Unfortunately there has been no `gradio_client` update/tag, only a (unrelated!) `@gradio/client` update, so this still won't build with a tag. I'll use the corresponding revision instead -- I've been patient enough :).

mdaniels5757 commented:
I guess this is a ZHF (#457852) fix now: Gradio has been failing on master the last couple weeks.

mdaniels5757 commented:
Actually this is convenient: because the build has never succeeded on 25.11, it's not a breaking change to update to v6. Standby...

mdaniels5757 committed:
python3Packages.gradio: 5.49.1 -> 6.1.0

Changelog: https://github.com/gradio-app/gradio/blob/gradio%406.1.0/CHANGELOG.md

mdaniels5757 committed:
python3Packages.gradio-pdf: 0.0.22 -> 0.0.23

Diff: https://github.com/freddyaboulton/gradio-pdf/compare/8833e9cd419d2a5eeff98e3ae8cbe690913bcfce...57deb0bd2823a9a75c5da498cc39a85f8fd4ff02

mdaniels5757 commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 446537`
Commit: `69696ae0444a091f4a967320016a6f7279723059`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 22 packages built:</summary>
  <ul>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio.dist</li>
    <li>python312Packages.smolagents</li>
    <li>python312Packages.smolagents.dist</li>
    <li>python312Packages.txtai</li>
    <li>python312Packages.txtai.dist</li>
    <li>python313Packages.gradio</li>
    <li>python313Packages.gradio-client</li>
    <li>python313Packages.gradio-client.dist</li>
    <li>python313Packages.gradio-pdf</li>
    <li>python313Packages.gradio-pdf.dist</li>
    <li>python313Packages.gradio.dist</li>
    <li>python313Packages.smolagents</li>
    <li>python313Packages.smolagents.dist</li>
    <li>python313Packages.txtai</li>
    <li>python313Packages.txtai.dist</li>
    <li>rembg</li>
    <li>rembg.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 20 packages built:</summary>
  <ul>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio.dist</li>
    <li>python312Packages.smolagents</li>
    <li>python312Packages.smolagents.dist</li>
    <li>python312Packages.txtai</li>
    <li>python312Packages.txtai.dist</li>
    <li>python313Packages.gradio</li>
    <li>python313Packages.gradio-client</li>
    <li>python313Packages.gradio-client.dist</li>
    <li>python313Packages.gradio-pdf</li>
    <li>python313Packages.gradio-pdf.dist</li>
    <li>python313Packages.gradio.dist</li>
    <li>python313Packages.smolagents</li>
    <li>python313Packages.smolagents.dist</li>
    <li>python313Packages.txtai</li>
    <li>python313Packages.txtai.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 16 packages built:</summary>
  <ul>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio.dist</li>
    <li>python312Packages.smolagents</li>
    <li>python312Packages.smolagents.dist</li>
    <li>python313Packages.gradio</li>
    <li>python313Packages.gradio-client</li>
    <li>python313Packages.gradio-client.dist</li>
    <li>python313Packages.gradio-pdf</li>
    <li>python313Packages.gradio-pdf.dist</li>
    <li>python313Packages.gradio.dist</li>
    <li>python313Packages.smolagents</li>
    <li>python313Packages.smolagents.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 16 packages built:</summary>
  <ul>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio.dist</li>
    <li>python312Packages.smolagents</li>
    <li>python312Packages.smolagents.dist</li>
    <li>python313Packages.gradio</li>
    <li>python313Packages.gradio-client</li>
    <li>python313Packages.gradio-client.dist</li>
    <li>python313Packages.gradio-pdf</li>
    <li>python313Packages.gradio-pdf.dist</li>
    <li>python313Packages.gradio.dist</li>
    <li>python313Packages.smolagents</li>
    <li>python313Packages.smolagents.dist</li>
  </ul>
</details>
